packages/utils-algebra/lib/utils.ts

```diff
@@ -78,18 +86,136 @@ const transformer = new TransformerSubTyped<KnownOperation>({
   [Types.COPY]: { ignoreKeys: new Set([ 'source', 'destination', 'metadata' ]) },
 });
 
-export const mapOperation: (typeof transformer.transformNode<'unsafe', Operation>) = <any>
-  transformer.transformNode.bind(transformer);
-
+/**
+ * Transform a single operation, similar to {@link mapOperation}, but using stricter typings. e.g.:
+ * `mapOperationStrict({type: 'distinct', input: {type: 'project' }}, { project: { transform: () => 5 } })`
+ * returns `{type: distinct, input: 5}`
+ * @param startObject the object from which we will start the transformation,
+ *   potentially visiting and transforming its descendants along the way.
+ * @param nodeCallBacks a dictionary mapping the various operation types to objects optionally
+ *    containing preVisitor and transformer.
+ *    The preVisitor allows you to provide {@link TransformContext} for the current object,
+ *    altering how it will be transformed.
+ *    The transformer allows you to manipulate the copy of the current object,
+ *    and expects you to return the value that should take the current objects place.
+ * @return the result of transforming the requested descendant operations (based on the preVisitor)
+ * using a transformer that works its way back up from the descendant to the startObject.
+ */
 export const mapOperationStrict = transformer.transformNode.bind(transformer);
 
-export const mapOperationSub: (typeof transformer.transformNodeSpecific<'unsafe', Operation>) = <any>
-  transformer.transformNodeSpecific.bind(transformer);
+/**
+ * Transform a single operation. e.g.:
+ * `mapOperation({type: 'distinct', input: {type: 'project' }}, { project: { transform: () => 5 } })`
```

Comment:
Shall we use a more realistic example?
Something similar to https://github.com/comunica/comunica/blob/next/major/packages/actor-optimize-query-operation-assign-sources-exhaustive/lib/ActorOptimizeQueryOperationAssignSourcesExhaustive.ts#L75-L88 for example.
Having a previsitor in there could also be useful.

Reply:
Sure, that works!

Two minor nits:
- Make sure that the example code is syntactically valid TS (e.g. `returns =`), because some people (including myself) may copy-paste that code.
- Can you modify the code so that indentation and newline placement is similar to other places in Comunica?